qucs/qucs-powercombining/qucspowercombiningtool.cpp

```diff
@@ -0,0 +1,1289 @@
+        BranchesCombo->addItem("7");
+        BranchesCombo->setEditable(true);
+    }
+    if(index == 6)// Travelling wave
+    {
+        BranchesCombo->clear();
+        BranchesCombo->addItem("3");
+        BranchesCombo->addItem("4");
+        BranchesCombo->addItem("5");
+        BranchesCombo->addItem("6");
+        BranchesCombo->setEditable(true);
+    }
+    if(index == 7)//Tree
+    {
+        BranchesCombo->clear();
+        BranchesCombo->addItem("4");
+        BranchesCombo->addItem("8");
+        BranchesCombo->addItem("16");
+        BranchesCombo->setEditable(true);
+    }
+}
+
+//---------------------------------------------------------------
+// This function reads the input values and calls the different methods for generating a schematic
+void QucsPowerCombiningTool::on_GenerateButton_clicked()
+{
+    double err=0;
+    double Z0 = RefImplineEdit->text().toDouble();
+    int N=BranchesCombo->currentText().toInt();
+    double K=K1lineEdit->text().toDouble();//Power ratio. Output port 1
+    double Freq = FreqlineEdit->text().toDouble()*getScaleFreq();
+    bool SP_block = AddSparcheckBox->isChecked();
+    tSubstrate Substrate;
+    bool microcheck = MicrostripcheckBox->isChecked();
+    if (microcheck)
+    {
+        Substrate.er=RelPermcomboBox->currentText().section("  ", 0, 0).toDouble();
+        Substrate.height=SubstrateHeightlineEdit->text().toDouble()*1e-3;
+        Substrate.thickness=ThicknesslineEdit->text().toDouble()*1e-6;
+        Substrate.maxWidth=MaxWidthlineEdit->text().toDouble()*1e-3;
+        Substrate.minWidth=MinWidthlineEdit->text().toDouble()*1e-3;
+        Substrate.resistivity=ResistivitylineEdit->text().toDouble();
+        Substrate.tand=tanDlineEdit->text().toDouble();
+        Substrate.roughness=RoughnesslineEdit->text().toDouble();
+    }
+
+    switch (TopoCombo->currentIndex())
+    {
+    case 0: // Wilkinson
+        (Wilkinson(Z0, Freq, K, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 1: // Tee
+        (Tee(Z0, Freq, K*K, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 2: //Branchline
+        (Branchline(Z0, Freq, K*K, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 3: //Double box branchline
+        (DoubleBoxBranchline(Z0, Freq, K*K, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 4: //Bagley
+        (Bagley(Z0, Freq, N, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 5: // Gysel
+        (Gysel(Z0, Freq, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 6: // Travelling wave
+        (TravellingWave(Z0, Freq, N, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    case 7: //Tree
+        (Tree(Z0, Freq, N, SP_block, microcheck, Substrate)) ? err = 0: err=1;
+        break;
+    }
+    !err ? statusBar->showMessage(tr("Ready! Use CTRL+V to paste the schematic"), 2000):
```

Comment:
`[picky mode on]`

why all these ` function() ? err = 0: err=1` ? You could have defined a variable `bool result_valid` and then used `result_valid = function()` for every case, checking `if result_valid ...` after the end of the `switch`.
Besides, I'm used to see (simple) functions returning `0` in case of success and a negative number in case of error, so you would have `err = function()` and a check `if (!err)` at the end.

`[picky mode off]`